### DISCUSSION TITLE
Encryption and Bearware account web login should be free for all

### DISCUSSION THREAD

**Discussion by ahmednaalik51** [BearWare/TeamTalk5#1518] [Ideas]
Hey to everyone.
I know I will probably get a lot of people who say to me that this should not be the case but understand me out first.
Most other softwares, such as Skype, or Teams, and other software even Whatsapp offer encryption by default without having to pay for it. I understand that the bearware companies was needs to make moneys, but Team Talk is very great application for talk to people as well. I think that the web login and pro server should be free for all.
There are so many people being hacked in the days now, and it's not good. Many people fear for security and I think security is a right, not something you should be forced to purchase. I think everyone has a right to security and it's not fair that you have to purchase it, when you could go use any other application and have security built right into it.
Does anyone else think that this also should be made free for every one? My main language is not English so I'm sorry for any grammer mistakes, but I got rid of as many as I could. But TeamTalk is truly the best audio quality software, and many people do not want to move away from this app, but it seriously lacks security, and lacks encryption and all that. Instead, you have to pay for that which I really don't think is right, because I think security should be made available to everyone, because everyone should have that rights.
If you don't agree than you can closed this topic if you wish, but I hope people will agree about it because it would be very great.
Have a great day.

**Comment by bear101** [comment]
@ahmednaalik51 Who should pay for app stores, code signing licenses, website hosting, official servers, test hardware, etc? If it's not TeamTalk users who should pay, then who is it?

**Reply by ahmednaalik51** [reply]
Hi @bear101 you are right, and I do apologize. I thought may be the SDK users pay for that. But please read the other comment from @Nardol maybe that explains it better.

**Comment by Nardol** [comment]
I totally agree a paid version allows TeamTalk to be financed and we, as users, cannot forget maintaining TeamTalk has a cost and should contribute as possible.

But for encryption in particular, I think it should be included for all version as it is in the client.
Today TLS is a standard and it would be really better if it could be the same in TeamTalk.
And in the actual implementation, the base is unfortunately missing: there is no host name check so even an autosigned certificate is valid. Having a warning for autosigned or when host name is not the certificate's common name or one of the alternate names.
So it is encrypted but with no check, having it pay could looks a little inccomplete.

Having free encryption would not make a paid version useless because more than bearware.dk web login, the paid version could also add TLS authentication for example. Sure, it need to be developed :wink:

And why not also accepting donations for TeamTalk if some users wish to contribute financially and the pro server cost could be too expensive for them?

**Reply by ahmednaalik51** [reply]
I think that also donations are better as well

**Comment by tech10** [comment]
I can agree with two things here in this discussion. First, I think SSL encryption should be available on every server for free, OpenSSL is free and open source, after all. Second, encrypting a server without checking certificates validity is as good as having no encryption at all. This validation could be done either via fingerprint validation, or the same manner in which website SSL certs are validated, that would make it more secure on the client end. Such as it is now, a man in the middle attack could easily be used to decrypt and intercept anything that's encrypted, without anyone on the client side knowing that anything had changed.

There are certain advantages that a bearware.dk web login will provide you, and I think the web login should remain a paid feature. If that's free, you'd have to consider the possible load on the server that someone has to pay for in order to keep the web login operational. More availability means more people using a paid feature, and if that paid feature uses external resources, there's definitely a reason to keep it non-free.

**Comment by computertech1012** [comment]
Hi,
Well, I whole-heartedly agree that bearware account should probably stay a payed feature, but for encryption it is free for most other platforms and should be free for TeamTalk as well.
However, I also understand @bear101 when he says that the TeamTalk users should pay for the option so that money can be going towards the official servers.
However, some people don't like using the official servers because there are sometimes people that are bothering the users, and there's even times that people get kicked off of the servers for no reason at all, but the owners of the servers do nothing about it.
This is why I do think that encryption should at least be free for all if possible. But I do agree that the devs need to keep TeamTalk going with money.
Is it possible for anyone to kick people off of the Encryption UK server? Because if not then we can just use that one safely.
Thanks for reading!
